src/utils/html.rs

```diff
@@ -1,62 +1,79 @@
-use crate::error::Result;
-use failure::err_msg;
-use kuchiki::traits::TendrilSink;
-use kuchiki::NodeRef;
+use crate::web::page::TemplateData;
+use lol_html::errors::RewritingError;
+use tera::Context;
 
-/// Extracts the contents of the `<head>` and `<body>` tags from an HTML document, as well as the
-/// classes on the `<body>` tag, if any.
-pub fn extract_head_and_body(html: &str) -> Result<(String, String, String)> {
-    let dom = kuchiki::parse_html().one(html);
+pub(crate) fn rewrite_lol(
+    html: &[u8],
+    ctx: Context,
+    templates: &TemplateData,
+) -> Result<Vec<u8>, RewritingError> {
```

Comment:
Maybe document what the return value *means*, that the `Vec<u8>` is the fully rendered HTML ready to be served

Reply:
Done, what do you think?

Reply:
Looks good, maybe add a note that this should/will be valid UTF-8, but we just haven't checked ourselves